utils/fasm/src/parameters.cpp

```diff
@@ -4,30 +4,32 @@
 namespace fasm {
 
 void Parameters::AddParameter(const std::string &eblif_parameter, const std::string &fasm_feature) {
-    auto ret = features_.insert(std::make_pair(eblif_parameter, FeatureParameter()));
-
-    VTR_ASSERT(ret.second);
-
-    ret.first->second.feature = fasm_feature;
-    ret.first->second.width = FeatureWidth(fasm_feature);
+    FeatureParameter fp;
+    fp.width = FeatureWidth(fasm_feature);
+    fp.feature = fasm_feature;
+    features_.insert(std::make_pair(eblif_parameter, fp));
 }
 
 std::string Parameters::EmitFasmFeature(const std::string &eblif_parameter, const std::string &value) {
-    auto iter = features_.find(eblif_parameter);
-    if(iter == features_.end()) {
-        return "";
+    std::ostringstream out;
+    int count = 0;
+    auto range = features_.equal_range(eblif_parameter);
+    for(auto i = range.first; i != range.second; ++i) {
+        // Parameter should have exactly the expected number of bits.
+        const FeatureParameter &fp = i->second;
+        if(value.size() != fp.width) {
+            vpr_throw(VPR_ERROR_OTHER,
+                      __FILE__, __LINE__, "When emitting FASM for parameter %s, expected width of %d got width of %d, value = \"%s\".",
+                      eblif_parameter.c_str(), fp.width, value.size(), value.c_str());
+        }
+        VTR_ASSERT(value.size() == fp.width);
+        if(count > 0) {
```

Comment:
Why not just always output a newline?

Reply:
Because it will cause lots of blank lines, one per call of this function, since the caller also emits a newline. I didn't want to disrupt the code too much by fixing that in some more elegant way.

Reply:
The parent code (https://github.com/SymbiFlow/vtr-verilog-to-routing/blob/master%2Bwip/utils/fasm/src/fasm.cpp#L662) should eat the space

Reply:
Done, and I checked that the extra newlines were being filtered out.